compiler/luci/service/src/Nodes/CircleRsqrt.test.cpp

```diff
@@ -31,3 +32,68 @@ TEST(CloneNodeTest, clone_Rsqrt)
   auto cloned_rsqrt = dynamic_cast<luci::CircleRsqrt *>(cloned);
   ASSERT_NE(nullptr, cloned_rsqrt);
 }
+
+TEST(ShapeRuleTest, rsqrt_dynamic_shape)
+{
+  luci::CircleInput input;
+  luci::CircleRsqrt rsqrt;
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  input.shape({1, 1, 3, 4});
+  input.shape_status(luci::ShapeStatus::VALID);
+  input.dim(0).unset();
+  input.dim(1).unset();
+
+  rsqrt.x(&input);
+
+  ASSERT_TRUE(shape_inf_rule.infer(&rsqrt, shape));
+  ASSERT_EQ(shape.rank(), 4);
+  ASSERT_FALSE(shape.dim(0).known());
+  ASSERT_FALSE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+
+  ASSERT_EQ(0, shape.dim(0).value());
+  ASSERT_EQ(0, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(4, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, rsqrt_static_shape)
+{
+  luci::CircleInput input;
+  luci::CircleRsqrt rsqrt;
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  input.shape({3, 2, 3, 4});
+  input.shape_status(luci::ShapeStatus::VALID);
+
+  rsqrt.x(&input);
+
+  ASSERT_TRUE(shape_inf_rule.infer(&rsqrt, shape));
+  ASSERT_EQ(shape.rank(), 4);
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_TRUE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+
+  ASSERT_EQ(3, shape.dim(0).value());
+  ASSERT_EQ(2, shape.dim(1).value());
+  ASSERT_EQ(3, shape.dim(2).value());
+  ASSERT_EQ(4, shape.dim(3).value());
+}
+
+TEST(ShapeRuleTest, rsqrt_shape_forward_NEG)
```

Comment:
we have a policy that keep more negative test case than positive.
please add more negative test case with `_NEG` suffix.

https://github.com/Samsung/ONE/pull/13735#discussion_r1725764564

Reply:
but rsqrt just forwards the shape, It might be hard to write more negative test cases.
so I think we need more tip about this case.

Reply:
@zetwhite, could you give us some advice?

Reply:
Thank you for pointing that out.
However, as @kyeong8139 mentioned, shape inference of rsqrt is not doing something logical. It just forwarding input shape to result shape. 

I think we need to get some advise.

Reply:
How about force to input unusual shape status. 😅
```c++
input.shape_status(luci::ShapeStatus::UNDEFINED);
```

Reply:
I have considered adding a NEG test like @kyeong8139 made.
However, it seems we need to add the same test for every 'just forwarding input shape to result shape' operation.
Is this the right strategy?

For now, I will add a test for the case of undefined status shape. Thank you!